cpg-core/src/main/java/de/fraunhofer/aisec/cpg/helpers/Benchmark.kt

```diff
@@ -25,15 +25,102 @@
  */
 package de.fraunhofer.aisec.cpg.helpers
 
+import de.fraunhofer.aisec.cpg.TranslationConfiguration
+import java.nio.file.Path
 import java.time.Duration
 import java.time.Instant
+import java.util.*
+import kotlin.IllegalArgumentException
 import org.slf4j.LoggerFactory
 
+/** Interface definition to hold different statistics about the translation process. */
+interface StatisticsHolder {
+    val translatedFiles: List<String>
+    val benchmarks: List<Benchmark>
+    val config: TranslationConfiguration
+
+    fun addBenchmark(b: Benchmark)
+
+    /** Pretty-prints benchmark results for easy copying to GitHub issues. */
+    fun printBenchmark() {
+        println("# Benchmark run ${UUID.randomUUID()}")
+        printMarkdown(
+            listOf(
+                listOf(
+                    "Translation config",
+                    "`${config.toString().replace(",", ", ").replace(":", ": ")}`"
+                ),
+                listOf("Number of files translated", translatedFiles.size),
+                listOf(
+                    "Translated file(s)",
+                    translatedFiles.map {
+                        relativeOrAbsolute(Path.of(it), config.topLevel.toPath())
+                    }
+                ),
+                *benchmarks
+                    .map { listOf("${it.caller}: ${it.message}", "${it.duration} ms") }
+                    .toTypedArray(),
+            ),
+            listOf("Metric", "Value")
+        )
+    }
+}
+
+/**
+ * Prints a table of values and headers in markdown format. Table columns are automatically adjusted
+ * to the longest column.
+ */
+fun printMarkdown(table: List<List<Any>>, headers: List<String>) {
+    val lengths = IntArray(headers.size)
+
+    // first, we need to calculate the longest column per line
+    for (row in table) {
+        for (i in row.indices) {
+            val value = row[i].toString()
+            if (value.length > lengths[i]) {
+                lengths[i] = value.length
+            }
+        }
+    }
+
+    // table header
+    val dash = lengths.joinToString(" | ", "| ", " |") { ("-".repeat(it)) }
+    var i = 0
+    val header = headers.joinToString(" | ", "| ", " |") { it.padEnd(lengths[i++]) }
+
+    // println(dash)
```

Comment:
I would remove this commented out code, if we want this dashes optional I would do so as a configuration. I don't know if the purpose was to keep this as an alternative or was just forgotten